### PR TITLE
Stop building hostmisc as a static lib.

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Contributes to https://github.com/dotnet/runtime/issues/1154

This broke the nethost from being consumable as a static lib.

See specific comment for context https://github.com/dotnet/runtime/issues/1154#issuecomment-618702097

The contained CMake changes are less than ideal and in most cases considered bad practice. I couldn't figure out a way to make this work without moving a fair bit of code and thus impacting higher priority single file work being done by @VSadov and @swaroop-sridhar.

/cc @vitek-karas

### CODE REVIEW

**Review by swaroop-sridhar — APPROVED** [review]
The change looks good. 
I think we should keep #1154 open until we have a test for it (to ensure libnethost.lib is correctly built).

**Review by VSadov — APPROVED** [review]
LGTM. Thanks!

File: src/installer/corehost/cli/lib.cmake [hunk 1]
```diff
@@ -5,13 +5,17 @@
...
 include(${CMAKE_CURRENT_LIST_DIR}/common.cmake)
+include(${CMAKE_CURRENT_LIST_DIR}/hostmisc/hostmisc.cmake)
 
 add_definitions(-D_NO_ASYNCRTIMP)
 add_definitions(-D_NO_PPLXIMP)
 add_definitions(-DEXPORT_SHARED_API=1)
 
+list(APPEND SOURCES
+    ${HOSTMISC_SRC}
+)
```
Comment (elinor-fung) [inline src/installer/corehost/cli/lib.cmake:16]: Is `libnethost` just relying on the fact that its cmake files includes `lib.cmake` before `lib_static.cmake`, so when the static library is added `SOURCES` is already updated?

File: src/installer/corehost/cli/exe.cmake [hunk 2]
```diff
@@ -16,10 +17,10 @@ include_directories(${CMAKE_CURRENT_LIST_DIR}/fxr)
 list(APPEND SOURCES
     ${CMAKE_CURRENT_LIST_DIR}/fxr_resolver.cpp
     ${CMAKE_CURRENT_LIST_DIR}/../corehost.cpp
+    ${HOSTMISC_SRC}
```
Comment (elinor-fung) [inline src/installer/corehost/cli/exe.cmake:20]: Should this file also add `HOSTMISC_HDR` to `HEADERS`?

File: src/installer/corehost/cli/test/nativehost/CMakeLists.txt [hunk 3]
```diff
@@ -45,6 +45,10 @@ if(CLR_CMAKE_TARGET_WIN32)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} /DELAYLOAD:nethost.dll")
 endif()
 
+include(../../hostmisc/hostmisc.cmake)
+list(APPEND SOURCES ${HOSTMISC_SRC})
+list(APPEND HEADERS ${HOSTMISC_HDR})
```
Comment (elinor-fung) [inline src/installer/corehost/cli/test/nativehost/CMakeLists.txt:50]: Can this go in `testexe.cmake` instead? I think pal/utils would probably be useful for any native test executables. I know `nativehost` is the only thing including `testexe.cmake` right now, but the intention is that `test_fx_ver` (and any future native test executables) will use it.

File: src/installer/corehost/cli/lib.cmake (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/installer/corehost/cli/lib.cmake:16]: Yep. I thought I tried that, but it does seem to work. Thanks.

File: src/installer/corehost/cli/exe.cmake (hunk 2 above)
Comment (AaronRobinsonMSFT) [reply src/installer/corehost/cli/exe.cmake:20]: No longer needed.

File: src/installer/corehost/cli/test/nativehost/CMakeLists.txt (hunk 3 above)
Comment (AaronRobinsonMSFT) [reply src/installer/corehost/cli/test/nativehost/CMakeLists.txt:50]: No needed.

**Pull request merged**